dev/tools/controllerbuilder/template/controller/controller.go

```diff
@@ -269,54 +259,70 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 	}
 
 	// TODO(user): Complete the gcp "UPDATE" or "PATCH" request with required fields.
-	req := &{{.KCCService}}pb.Update{{.Kind}}Request{
-		Resource:               resource,
-		Project:                a.ProjectID,
+	req := &{{.KCCService}}pb.Update{{.ProtoResource}}Request{
+		Name:       			a.id.FullyQualifiedName(),
+		UpdateMask:             updateMask,
+		{{.ProtoResource}}:     resource,
 	}
-	op, err := a.gcpClient.Update{{.Kind}}(ctx, req)
+	op, err := a.gcpClient.Update{{.ProtoResource}}(ctx, req)
 	if err != nil {
-		return fmt.Errorf("updating {{.Kind}} %s: %w", a.id.fullyQualifiedName(), err)
+		return fmt.Errorf("updating {{.ProtoResource}} %s: %w", a.id.FullyQualifiedName(), err)
 	}
 	// TODO(user): Adjust the response, depending on the LRO or not.
 	updated, err := op.Wait(ctx)
 	if err != nil {
-		return fmt.Errorf("{{.Kind}} %s waiting update: %w", a.id.fullyQualifiedName(), err)
+		return fmt.Errorf("{{.ProtoResource}} %s waiting update: %w", a.id.FullyQualifiedName(), err)
 	}
-	log.V(2).Info("successfully updated {{.Kind}}", "name", a.id.fullyQualifiedName())
+	log.V(2).Info("successfully updated {{.ProtoResource}}", "name", a.id.FullyQualifiedName())
 
 	status := &krm.{{.Kind}}Status{}
 	// TODO(user): (Optional) Please add the StatusObservedState_FromProto mappers under the same package
-	status := {{.Kind}}StatusObservedState_FromProto(mapCtx, created)
+	status := {{.Kind}}StatusObservedState_FromProto(mapCtx, updated)
 	if mapCtx.Err() != nil {
 		return mapCtx.Err()
 	}
 	return setStatus(u, status)
 }
 
 func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error) {
-	// TODO(kcc) 
-	return nil, nil
+	if a.actual == nil {
+		return nil, fmt.Errorf("Find() not called")
+	}
+	u := &unstructured.Unstructured{}
+
+	obj := &krm.{{.Kind}}{}
+	mapCtx := &direct.MapContext{}
+	obj.Spec = direct.ValueOf({{.Kind}}Spec_FromProto(mapCtx, a.actual))
+	if mapCtx.Err() != nil {
+		return nil, mapCtx.Err()
+	}
+	// TODO(user): Update other resource reference 
+	obj.Spec.ProjectRef = &refs.ProjectRef{Name: a.id.Parent.Project}
+	obj.Spec.Location = a.id.Parent.Location
+	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+	u.Object = uObj
+	return u, nil
 }
 
 // Delete implements the Adapter interface.
 func (a *Adapter) Delete(ctx context.Context, deleteOp *directbase.DeleteOperation) (bool, error) {
 	log := klog.FromContext(ctx).WithName(ctrlName)
-	log.V(2).Info("deleting {{.Kind}}", "name", a.id.fullyQualifiedName())
+	log.V(2).Info("deleting {{.ProtoResource}}", "name", a.id.FullyQualifiedName())
 
-	if a.resourceID == "" {
-		return false, nil
-	}
```

Comment:
This is actually the reason about this PR. Current controllers which use the old template shall fix it.

Reply:
I think this is OK? In the template, `a.resourceID` is set from either `spec.ResourceID` or `metadata.Name`.

Reply:
You are right. I was fixing here and there and don't know where this comment comes from